File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution(object):

    def digSum(self, n):
        total = 0

        while n:
            total += n % 10
            n //= 10

        return total

    def digProduct(self, n):
        product = 1

        while n:
            product *= n % 10
            n //= 10

        return product

    def checkDivisibility(self, n):
        a = self.digSum(n)
        b = self.digProduct(n)

        return n % (a + b) == 0

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int product =1,sum =0;
        int a =n;

        while(a){
            sum += a%10;
            product *= a%10;
            a=a/10;
        }

        return n%(sum+product)==0;
        

    }
};